src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -43,9 +43,12 @@ public function hashPassword($user, string $plainPassword): string
             trigger_deprecation('symfony/password-hasher', '5.3', 'The "%s()" method expects a "%s" instance as first argument. Not implementing it in class "%s" is deprecated.', __METHOD__, PasswordAuthenticatedUserInterface::class, get_debug_type($user));
         }
 
-        $salt = $user->getSalt();
-        if ($salt && !$user instanceof LegacyPasswordAuthenticatedUserInterface) {
-            trigger_deprecation('symfony/password-hasher', '5.3', 'Returning a string from "getSalt()" without implementing the "%s" interface is deprecated, the "%s" class should implement it.', LegacyPasswordAuthenticatedUserInterface::class, get_debug_type($user));
+        $salt = null;
+        if (method_exists($user, 'getSalt')) {
```

Comment:
If we keep support for that case (see my comment on the issue), I'd rather rewrite the code like that:

```php
$salt = null;

if ($user instanceof LegacyPasswordAuthenticatedUserInterface) {
    $salt = $user->getSalt();
} elseif ($user instanceof UserInterface) {
    $salt = $user->getSalt();

    if ($salt) {
        trigger_deprecation(...);
    }
}
```

this way, the code will support `getSalt` either for the new LegacyPasswordAuthenticatedUserInterface or when using UserInterface (i.e. the pre-5.3 API) and the code will be more similar to the 6.0 code (which has this `if` check but not the `elseif`) which simplifies merging branches

Reply:
Works for me.

Reply:
done